tests/test_model_graph.py

```diff
@@ -473,3 +481,56 @@ def test_custom_node_formatting_graphviz(simple_model):
         ]
     )
     assert body == items
+
+
+def test_none_dim_in_plate() -> None:
+    coords = {
+        "obs": range(5),
+    }
+    with pm.Model(coords=coords) as model:
+        data = pt.as_tensor_variable(
+            np.ones((5, 3)),
+            name="data",
+        )
+        pm.Deterministic("C", data, dims=("obs", None))
+
+    graph = ModelGraph(model)
+
+    assert graph.get_plates() == [
+        Plate(
+            dim_info=DimInfo(names=("obs", None), sizes=(5, 3)),
+            variables=[NodeInfo(var=model["C"], node_type=NodeType.DETERMINISTIC)],
+        ),
+    ]
+    assert graph.edges() == []
+
+
+def test_shape_without_dims() -> None:
+    with pm.Model() as model:
+        pm.Normal("mu", shape=5)
+
+    graph = ModelGraph(model)
+
+    assert graph.get_plates() == [
+        Plate(
+            dim_info=DimInfo(names=(None,), sizes=(5,)),
+            variables=[NodeInfo(var=model["mu"], node_type=NodeType.FREE_RV)],
+        ),
+    ]
+    assert graph.edges() == []
+
+
+def test_scalars_have_no_dim_info() -> None:
+    with pm.Model() as model:
+        pm.Normal("x")
+
+    graph = ModelGraph(model)
+
+    assert graph.get_plates() == [
+        Plate(
+            dim_info=None,
```

Comment:
I prefer your first approach first, where it's a DimInfo with empty names and sizes (which can evaluate to False as well)
```suggestion
            dim_info=None,
```